### PR TITLE
=per #18342 Improve docs for leveldb plugin config

### PR DESCRIPTION
**Pull request by patriknw**
- clarify how to enable the plugin
- added empty class property in fallback config in reference
  to have a proper place to document that and throw a more
  specific exception if it is not defined
- also some formatting of reference.conf

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #18342

**Comment by patriknw** [comment]
Please review @ktoso

File: akka-docs/rst/java/persistence.rst [hunk 1]
```diff
@@ -859,7 +876,11 @@ Local snapshot store
 --------------------
 
 Local snapshot store plugin config entry is ``akka.persistence.snapshot-store.local`` and it writes snapshot files to
-the local filesystem. The default storage location is a directory named ``snapshots`` in the current working
+the local filesystem. Enable this plugin by defining config property:
```
Comment (ktoso) [inline akka-docs/rst/java/persistence.rst]: Maybe add an explicit note that it's OK to run without a snapshot store configured even if you use persistence, until you actually call saveSnapshot somewhere (and that cluster sharding will use snapshotting?).

File: akka-persistence/src/main/resources/reference.conf [hunk 2]
```diff
@@ -5,32 +5,29 @@
...
-#           The class must have a constructor without parameters or constructor with 
-#           one `com.typesafe.config.Config` parameter.
-# `plugin-dispatcher` : Absolute configuration path to the akka dispatcher configuration entry. This string field is optional.  
-
-# Note that journal and snapshot store plugins included with the extension are suitable for testing purposes only.
-# You should change extension defaults or override `journalPluginId` and `snapshotPluginId` in the persistent actor or view.
-# Directory of persistence journal and snapshot store plugins is available at the Akka Community Projects page 
-# http://akka.io/community/
+# Directory of persistence journal and snapshot store plugins is available at the 
+# Akka Community Projects page http://akka.io/community/
```
Comment (ktoso) [inline akka-persistence/src/main/resources/reference.conf]: good idea

File: akka-persistence/src/main/scala/akka/persistence/Persistence.scala [hunk 3]
```diff
@@ -253,7 +253,11 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
 
   private def createPlugin(configPath: String, pluginConfig: Config): ActorRef = {
     val pluginActorName = configPath
-    val pluginClassName = pluginConfig.getString("class")
+    val pluginClassName = pluginConfig.getString("class") match {
+      case "" ⇒ throw new IllegalArgumentException("Plugin class name must be defined in config property " +
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/Persistence.scala]: good catch

**Comment by ktoso** [comment]
LGTM, maybe add the note about snapshot store unless it's mentioned somewhere already

**Comment by patriknw** [comment]
I added some more notes about that snapshot store is optional

File: akka-docs/rst/scala/persistence.rst [hunk 4]
```diff
@@ -516,6 +516,8 @@ saved snapshot matches the specified ``SnapshotSelectionCriteria`` will replay a
   Since it is acceptable for some applications to not use any snapshotting, it is legal to not configure a snapshot store,
   however Akka will log a warning message when this situation is detected and then continue to operate until
   an actor tries to store a snapshot, at which point the the operation will fail (by replying with an ``SaveSnapshotFailure`` for example).
+  
+  Note that :ref:`cluster_sharding_scala` is using snapshots, so if you use Cluster Sharding you need to define a snapshot store plugin.
```
Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: Thanks :)

**Pull request merged**